cmd/flags/admin.go

```diff
@@ -70,13 +76,26 @@ func (s *AdminServer) setLogger(logger *zap.Logger) {
 // AddFlags registers CLI flags.
 func (s *AdminServer) AddFlags(flagSet *flag.FlagSet) {
 	flagSet.String(adminHTTPHostPort, s.adminHostPort, fmt.Sprintf("The host:port (e.g. 127.0.0.1%s or %s) for the admin server, including health check, /metrics, etc.", s.adminHostPort, s.adminHostPort))
+	tlsAdminHTTPFlagsConfig.AddFlags(flagSet)
 }
 
 // InitFromViper initializes the server with properties retrieved from Viper.
-func (s *AdminServer) initFromViper(v *viper.Viper, logger *zap.Logger) {
+func (s *AdminServer) initFromViper(v *viper.Viper, logger *zap.Logger) error {
 	s.setLogger(logger)
 
 	s.adminHostPort = v.GetString(adminHTTPHostPort)
+	if tlsAdminHTTP, err := tlsAdminHTTPFlagsConfig.InitFromViper(v); err == nil {
+		s.TLSConfig = tlsAdminHTTP
+	} else {
+		return fmt.Errorf("failed to parse admin server TLS options: %w", err)
+	}
+	if s.TLSConfig.Enabled {
+		// This checks if the certificates are correctly provided
+		if _, err := s.TLSConfig.Config(s.logger); err != nil {
```

Comment:
why not just store the returned config similar to `tlsCertWatcherCloser`?

Reply:
Done :) !